sklearn/model_selection/_search.py

```diff
@@ -724,7 +731,8 @@ def evaluate_candidates(candidate_params):
 
         return self
 
-    def _format_results(self, candidate_params, scorers, n_splits, out):
+    def _format_results(self, candidate_params, scorers, n_splits, out,
+                        more_results={}):
```

Comment:
`more_results` can be a regular argument (not a keyword).

Reply:
We'd have to change `GridSearchCV` and `RandomizedSearchCV` then. Would you prefer it?

Reply:
> I also prefer the  private API to have as few params with default values as possible

Why? I'm not sold on this

I like to not break backward compatibility when we can

Reply:
It can also typically make writing test annoying (see e.g. `_find_binning_thresholds` in sklearn/ensemble/_hist_gradient_boosting/tests/test_binning.py)

Reply:
> I avoid defaults for private utilities, because I have seen a history of bugs caused by parameters not being passed when they should, and that bug not being identified with thanks to a default value

I'm not sure how this specifically applies to private utilities? It seems to me that this holds for any function, whether it's private or public.

Would we all be happy with this? (adding the kwonly part)

```py
    def _format_results(self, candidate_params, *, scorers, n_splits, out,
                        more_results=None):
```

Reply:
It applies to private utilities because there is little usability benefit to providing default values when the utility is private. Yes, I'm fine with that in this case.